cmd/root.go

```diff
@@ -226,6 +228,17 @@ func parseConfig(cmd *Command, conf *proxy.Config, args []string) error {
 	if conf.CredentialsFile != "" && conf.GcloudAuth {
 		return newBadCommandError("cannot specify --credentials-file and --gcloud-auth flags at the same time")
 	}
+	if userHasSet("admin-api-endpoint") {
+		_, err := url.Parse(conf.APIEndpointURL)
+		if err != nil {
+			return newBadCommandError(fmt.Sprintf("provided value for --alloydbadmin-api-endpoint is not a valid url, %v", conf.APIEndpointURL))
+		}
+
+		// Remove trailing '/' if included
+		if strings.HasSuffix(conf.APIEndpointURL, "/") {
+			conf.APIEndpointURL = strings.TrimSuffix(conf.APIEndpointURL, "/")
+		}
```

Comment:
Let's add the log statement here, since always logging the endpoint isn't useful for most users.